src/components/TextInput/TextInput.tsx

```diff
@@ -54,16 +56,23 @@ export const TextInput = React.forwardRef<HTMLSpanElement, TextInputProps>(funct
         style,
         className,
         qa,
+        controlProps: controlPropsOriginal,
+        labelProps,
     } = props;
     const [uncontrolledValue, setUncontrolledValue] = React.useState(defaultValue ?? '');
     const innerControlRef = React.useRef<HTMLTextAreaElement | HTMLInputElement>(null);
+    const labelRef = React.useRef<HTMLSpanElement>(null);
+    const [labelWidth, setLabelWidth] = React.useState(0);
     const [hasVerticalScrollbar, setHasVerticalScrollbar] = React.useState(false);
 
     const isControlled = value !== undefined;
     const inputValue = isControlled ? value : uncontrolledValue;
+    const isLabelVisible = !multiline && Boolean(label);
 
     const handleRef = useForkRef(props.controlRef, innerControlRef);
 
+    const innerControlSize = useElementSize(isLabelVisible ? innerControlRef : null);
```

Comment:
Can we get label size here?

Reply:
yes, looks like `labelSize` / `labelRef` also ok. will change.